.github/workflows/Deploy.yaml

```diff
@@ -27,49 +31,99 @@ on:
+  contents: write
+  actions: read
+
+concurrency:
+  group: 'deploy'
+  cancel-in-progress: true
 
 defaults:
   run:
     shell: pwsh
 
 jobs:
+  Inputs:
+    runs-on: [ ubuntu-latest ]
+    outputs:
+      branch: ${{ steps.CreateInputs.outputs.branch }}
+      copyToMain: ${{ steps.CreateInputs.outputs.copyToMain }}
+      directCommit: ${{ steps.CreateInputs.outputs.directCommit }}
+      requireEndToEndTests: ${{ steps.CreateInputs.outputs.requireEndToEndTests }}
+      createRelease: ${{ steps.CreateInputs.outputs.createRelease }}
+      defaultBcContainerHelperVersion: ${{ steps.CreateInputs.outputs.defaultBcContainerHelperVersion }}
+    steps:
+      - name: Create inputs
+        id: CreateInputs
+        run: |
+          if ($env:GITHUB_EVENT_NAME -eq 'workflow_run') {
+            $branch = 'preview'
+            $copyToMain = 'false'
+            $directCommit = 'false'
+            $requireEndToEndTests = 'true'
+            $createRelease = 'false'
+            $defaultBcContainerHelperVersion = 'preview'
+          } else {
+            $branch = '${{ github.event.inputs.branch }}'
+            $copyToMain = '${{ github.event.inputs.copyToMain }}'
+            $directCommit = '${{ github.event.inputs.directCommit }}'
+            $requireEndToEndTests = '${{ github.event.inputs.requireEndToEndTests }}'
+            $createRelease = '${{ github.event.inputs.createRelease }}'
+            $defaultBcContainerHelperVersion = '${{ github.event.inputs.defaultBcContainerHelperVersion }}'
+          }
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "branch=$branch"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "copyToMain=$copyToMain"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "directCommit=$directCommit"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "requireEndToEndTests=$requireEndToEndTests"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "createRelease=$createRelease"
+          Add-Content -encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "defaultBcContainerHelperVersion=$defaultBcContainerHelperVersion"
+
   CheckEndToEnd:
     runs-on: [ ubuntu-latest ]
+    needs: [ Inputs ]
     steps:
       - name: Check successful end 2 end tests have run
-        if: github.repository_owner == 'microsoft' && github.event.inputs.requireEndToEndTests == 'true'
+        if: github.repository_owner == 'microsoft' && needs.Inputs.outputs.requireEndToEndTests == 'true'
         env:
           GH_TOKEN: ${{ github.token }}
         run: |
           $errorActionPreference = "Stop"
-          $end2endRuns = (gh api /repos/$($env:GITHUB_REPOSITORY)/actions/runs?per_page=100 | ConvertFrom-Json).workflow_runs | Where-Object { $_.name -eq 'End 2 End Tests' }
+          $end2endRuns = (gh api /repos/$($env:GITHUB_REPOSITORY)/actions/runs?per_page=100 | ConvertFrom-Json).workflow_runs | Where-Object { $_.name -eq 'End to end tests' }
```

Comment:
[List workflow runs for a workflow](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-workflow) is better suited here.